### PR TITLE
Add the support of RESP3 double type

### LINKED ISSUE #1995: Add RESP3 double type
Please refer [Redis RESP3](https://github.com/redis/redis-specifications/blob/master/protocol/RESP3.md)

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #1995

### CODE REVIEW

File: src/commands/cmd_hash.cc [hunk 1]
```diff
@@ -372,7 +372,13 @@ class CommandHScan : public CommandSubkeyScanBase {
       return {Status::RedisExecErr, s.ToString()};
     }
 
-    *output = GenerateOutput(srv, conn, fields, values, CursorType::kTypeHash);
+    auto cursor = GetNextCursor(srv, fields, CursorType::kTypeHash);
+    std::vector<std::string> entries;
```
Comment (torwig) [inline src/commands/cmd_hash.cc:376]: Can we utilize the `entries.reserve()` call here?

File: src/commands/cmd_zset.cc [hunk 2]
```diff
@@ -1350,12 +1351,13 @@ class CommandZScan : public CommandSubkeyScanBase {
       return {Status::RedisExecErr, s.ToString()};
     }
 
-    std::vector<std::string> score_strings;
-    score_strings.reserve(scores.size());
-    for (const auto &score : scores) {
-      score_strings.emplace_back(util::Float2String(score));
+    auto cursor = GetNextCursor(srv, members, CursorType::kTypeHash);
+    std::vector<std::string> entries;
```
Comment (torwig) [inline src/commands/cmd_zset.cc:1355]: And `reserve` here as well?

File: src/commands/cmd_zset.cc [hunk 3]
```diff
@@ -1350,12 +1351,13 @@ class CommandZScan : public CommandSubkeyScanBase {
       return {Status::RedisExecErr, s.ToString()};
     }
 
-    std::vector<std::string> score_strings;
-    score_strings.reserve(scores.size());
-    for (const auto &score : scores) {
-      score_strings.emplace_back(util::Float2String(score));
+    auto cursor = GetNextCursor(srv, members, CursorType::kTypeHash);
```
Comment (jihuayu) [inline src/commands/cmd_zset.cc]:
```suggestion
    auto cursor = GetNextCursor(srv, members, CursorType::kTypeZSet);
```

**Review by jihuayu — COMMENTED** [review]
Thanks. Only a little typo, Others is LGTM.

File: src/commands/cmd_zset.cc (hunk 3 above)
Comment (git-hulk) [reply src/commands/cmd_zset.cc]: Good catch.

**Comment by git-hulk** [comment]
@jihuayu @torwig Thanks for your good reviews. All are done, plz take a look again.

**Review by jihuayu — APPROVED** [review]

**Review by caipengbo — APPROVED** [review]

**Pull request merged**